src/nvim/api/buffer.c

```diff
@@ -886,7 +886,7 @@ Object nvim_buf_get_var(Buffer buffer, String name, Error *err)
   buf_T *buf = find_buffer_by_handle(buffer, err);
 
   if (!buf) {
-    return (Object)OBJECT_INIT;
+    return NIL;
```

Comment:
This is quite a broad macro name. Could we rename this to `OBJECT_NIL` or `NIL_OBJECT`?

Reply:
I am in agreement with this. `NIL` is pretty vague. @bfredl thoughts?

Reply:
I prefer keeping `OBJECT_INIT`. It's more consistent with other macros.

Reply:
`OBJECT_NIL` is more consistent with `OBJECT_INIT`. `NIL_OBJ` would be consistent with `BOOLEAN_OBJ`.

Reply:
> I prefer keeping `OBJECT_INIT`. It's more consistent with other macros.

That's also an option. But I would prefer `NIL_OBJ`

Reply:
There's also many more instances of `(Object)OBJECT_INIT` (currently `NIL`) than `OBJECT_INIT` by itself, another factor to consider.

Reply:
Ah, seems like we need* `OBJECT_INIT` to initialize static `Object`s. We can manually provide `{ .type = kObjectTypeNil }` initializer but `OBJECT_INIT` is clearer. Regardless, I still think `NIL` should be renamed to `NIL_OBJ` if we want to keep it, or remove `NIL` entirely and replace it with `(Object)OBJECT_INIT`

Reply:
So just leave `OBJECT_INIT` alone and change `NIL` and `(Object)OBJECT_INIT` to `NIL_OBJ`.

Reply:
Alright

Reply:
changing `NIL` to `NIL_OBJ` sounds like like unnecessary churn. It is fine as it is.

Reply:
alright then